app-i18n/translate-shell: bump version to 0.9.6.11

ferki opened the pull request:
Most notably, this release adds compatibility with sys-apps/gawk-5*.

Please review and merge, or let me know if it needs any further changes.

Closes: https://bugs.gentoo.org/691626
Package-Manager: Portage-2.3.71, Repoman-2.3.17

juippis commented on app-i18n/translate-shell/translate-shell-0.9.6.11.ebuild:
```diff
@@ -0,0 +1,37 @@
+RDEPEND="
+	app-misc/rlwrap
+	sys-apps/gawk
+	curl? ( net-misc/curl[ssl] )
```
These use flags doesn't seem to be doing anything. USE-controlled rdepends is forbidden per https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Policies#USE-Controlled_Optional_RDEPENDs

If they have no function during configure or compile, they should be moved to `optfeature` or similar instead.

ferki commented:
Thanks for the patience. I have rebased my branch on top of current master, and then changed the ebuild to use `optfeature` for optional RDEPENDs.

There's one thing I don't fully like about it and would appreciate hints on: if all the optional dependencies are already installed, it still prints the initial `You may need to install the following optional packages for full functionality:` line, without listing anything afterwards. What would be the best way around that?

juippis commented:
> There's one thing I don't fully like about it and would appreciate hints on: if all the optional dependencies are already installed, it still prints the initial `You may need to install the following optional packages for full functionality:` line, without listing anything afterwards. What would be the best way around that?

`optfeature` is quite fancy because it only prints packages that are missing from target system. So if you want to leave that line in the ebuild, the best you can do is word it to make it obvious that the following list ONLY has packages NOT CURRENTLY INSTALLED. I think yours is pretty good at that already.

juippis commented:
```
>>> Source compiled.
>>> Test phase: app-i18n/translate-shell-0.9.6.11
make -j16 -l16 NETWORK_ACCESS=no test 
[OK] Task build completed.
====== TESTING FOR Commons.awk STARTED
[PASS] ( 1/3 ) TESTING anything()
[PASS] ( 2/3 ) TESTING anything()
[PASS] ( 3/3 ) TESTING anything()
[PASS] ( 1/4 ) TESTING exists()
[PASS] ( 2/4 ) TESTING exists()
[PASS] ( 3/4 ) TESTING exists()
[PASS] ( 4/4 ) TESTING exists()
[PASS] ( 1/3 ) TESTING belongsTo()
[PASS] ( 2/3 ) TESTING belongsTo()
[PASS] ( 3/3 ) TESTING belongsTo()
[PASS] ( 1/4 ) TESTING identical()
[PASS] ( 2/4 ) TESTING identical()
[PASS] ( 3/4 ) TESTING identical()
[PASS] ( 4/4 ) TESTING identical()
[PASS] ( 1/1 ) TESTING append()
[PASS] ( 1/4 ) TESTING isnum()
[PASS] ( 2/4 ) TESTING isnum()
[PASS] ( 3/4 ) TESTING isnum()
[PASS] ( 4/4 ) TESTING isnum()
[PASS] ( 1/3 ) TESTING startsWithAny()
[PASS] ( 2/3 ) TESTING startsWithAny()
[PASS] ( 3/3 ) TESTING startsWithAny()
[PASS] ( 1/4 ) TESTING matchesAny()
[PASS] ( 2/4 ) TESTING matchesAny()
[PASS] ( 3/4 ) TESTING matchesAny()
[PASS] ( 4/4 ) TESTING matchesAny()
[PASS] ( 1/4 ) TESTING replicate()
[PASS] ( 2/4 ) TESTING replicate()
[PASS] ( 3/4 ) TESTING replicate()
[PASS] ( 4/4 ) TESTING replicate()
[PASS] ( 1/4 ) TESTING reverse()
[PASS] ( 2/4 ) TESTING reverse()
[PASS] ( 3/4 ) TESTING reverse()
[PASS] ( 4/4 ) TESTING reverse()
[PASS] ( 1/4 ) TESTING join()
[PASS] ( 2/4 ) TESTING join()
[PASS] ( 3/4 ) TESTING join()
[PASS] ( 4/4 ) TESTING join()
[PASS] ( 1/1 ) TESTING explode()
[PASS] ( 1/8 ) TESTING escapeChar()
[PASS] ( 2/8 ) TESTING escapeChar()
[PASS] ( 3/8 ) TESTING escapeChar()
[PASS] ( 4/8 ) TESTING escapeChar()
[PASS] ( 5/8 ) TESTING escapeChar()
[PASS] ( 6/8 ) TESTING escapeChar()
[PASS] ( 7/8 ) TESTING escapeChar()
[PASS] ( 8/8 ) TESTING escapeChar()
[PASS] ( 1/6 ) TESTING literal()
[PASS] ( 2/6 ) TESTING literal()
[PASS] ( 3/6 ) TESTING literal()
[PASS] ( 4/6 ) TESTING literal()
[PASS] ( 5/6 ) TESTING literal()
[PASS] ( 6/6 ) TESTING literal()
[PASS] ( 1/4 ) TESTING escape()
[PASS] ( 2/4 ) TESTING escape()
[PASS] ( 3/4 ) TESTING escape()
[PASS] ( 4/4 ) TESTING escape()
[PASS] ( 1/4 ) TESTING unescape()
[PASS] ( 2/4 ) TESTING unescape()
[PASS] ( 3/4 ) TESTING unescape()
[PASS] ( 4/4 ) TESTING unescape()
[PASS] ( 1/10) TESTING parameterize()
[PASS] ( 2/10) TESTING parameterize()
[PASS] ( 3/10) TESTING parameterize()
[PASS] ( 4/10) TESTING parameterize()
[PASS] ( 5/10) TESTING parameterize()
[PASS] ( 6/10) TESTING parameterize()
[PASS] ( 7/10) TESTING parameterize()
[PASS] ( 8/10) TESTING parameterize()
[PASS] ( 9/10) TESTING parameterize()
[PASS] (10/10) TESTING parameterize()
[PASS] ( 1/10) TESTING unparameterize()
[PASS] ( 2/10) TESTING unparameterize()
[PASS] ( 3/10) TESTING unparameterize()
[PASS] ( 4/10) TESTING unparameterize()
[PASS] ( 5/10) TESTING unparameterize()
[PASS] ( 6/10) TESTING unparameterize()
[PASS] ( 7/10) TESTING unparameterize()
[PASS] ( 8/10) TESTING unparameterize()
[PASS] ( 9/10) TESTING unparameterize()
[PASS] (10/10) TESTING unparameterize()
[PASS] ( 1/4 ) TESTING toString()
[PASS] ( 2/4 ) TESTING toString()
[PASS] ( 3/4 ) TESTING toString()
[PASS] ( 4/4 ) TESTING toString()
[PASS] ( 1/4 ) TESTING squeeze()
[PASS] ( 2/4 ) TESTING squeeze()
[PASS] ( 3/4 ) TESTING squeeze()
[PASS] ( 4/4 ) TESTING squeeze()
[PASS] ( 1/12) TESTING yn()
[PASS] ( 2/12) TESTING yn()
[PASS] ( 3/12) TESTING yn()
[PASS] ( 4/12) TESTING yn()
[PASS] ( 5/12) TESTING yn()
[PASS] ( 6/12) TESTING yn()
[PASS] ( 7/12) TESTING yn()
[PASS] ( 8/12) TESTING yn()
[PASS] ( 9/12) TESTING yn()
[PASS] (10/12) TESTING yn()
[PASS] (11/12) TESTING yn()
[PASS] (12/12) TESTING yn()
[PASS] ( 1/4 ) TESTING quote()
[PASS] ( 2/4 ) TESTING quote()
[PASS] ( 3/4 ) TESTING quote()
[PASS] ( 4/4 ) TESTING quote()
[PASS] ( 1/6 ) TESTING unquote()
[PASS] ( 2/6 ) TESTING unquote()
[PASS] ( 3/6 ) TESTING unquote()
[PASS] ( 4/6 ) TESTING unquote()
[PASS] ( 5/6 ) TESTING unquote()
[PASS] ( 6/6 ) TESTING unquote()
[PASS] ( 1/4 ) TESTING fileExists()
[PASS] ( 2/4 ) TESTING fileExists()
[PASS] ( 3/4 ) TESTING fileExists()
[PASS] ( 4/4 ) TESTING fileExists()
[PASS] ( 1/4 ) TESTING dirExists()
[PASS] ( 2/4 ) TESTING dirExists()
[PASS] ( 3/4 ) TESTING dirExists()
[PASS] ( 4/4 ) TESTING dirExists()
[PASS] ( 1/2 ) TESTING detectProgram()
[PASS] ( 2/2 ) TESTING detectProgram()
[PASS] (121/121) TESTING COMPLETED. PERFECT!
====== TESTING FOR Utils.awk STARTED
[PASS] ( 1/1 ) TESTING GawkVersion
[FAIL] ( 1/1 ) TESTING Rlwrap
       assertEqual:  IS NOT EQUAL TO rlwrap
[PASS] ( 1/1 ) TESTING Emacs
[PASS] ( 1/5 ) TESTING newerVersion()
[PASS] ( 2/5 ) TESTING newerVersion()
[PASS] ( 3/5 ) TESTING newerVersion()
[PASS] ( 4/5 ) TESTING newerVersion()
[PASS] ( 5/5 ) TESTING newerVersion()
[PASS] ( 1/3 ) TESTING dump()
[PASS] ( 2/3 ) TESTING dump()
[PASS] ( 3/3 ) TESTING dump()
[PASS] ( 1/2 ) TESTING dumpX()
[PASS] ( 2/2 ) TESTING dumpX()
[PASS] ( 1/1 ) TESTING base64()
[PASS] ( 1/1 ) TESTING uprintf
[FAIL] (14/15) TESTING COMPLETED
       14 PASSED, 1 FAILED
make: *** [Makefile:26: test] Error 1
 * ERROR: app-i18n/translate-shell-0.9.6.11::testworld failed (test phase):
 *   emake failed
```

Any ideas about that?

ferki commented:
@juippis: thanks for the feedback! I've rebased my changes on top of current master, and updated the following things in this iteration:
 - added `RESTRICT="!test? ( test )"` to address the QA warning from above
 - replaced virtual/emacs with app-editors/emacs in test dependencies to keep up with the recent changes around emacs ebuilds in tree
 - since the test suite of translate-shell tests for rlwrap and emacs in pretty much the same way, I've decided to include both as test dependencies
 - since "rlwrap or emacs" is a dependency for the same "interactive translation" feature, I've updated the relevant optfeature line accordingly

Hopefully all these combined means we'll end up with a greatly improved ebuild, but feel free to let me know if I can do more!

ferki commented:
Thank you, @juippis!
